src/templates/next/components/Navbar/Navbar.tsx

```diff
@@ -1,178 +1,285 @@
+                )
+              }
+
+              return (
+                <li key={Math.random()} className="inline">
+                  <button
+                    className={`px-3 py-4 text-lg text-content-medium transition ease-in-out duration-300 hover:bg-interaction-sub active:bg-interaction-sub ${
+                      currentPathName.startsWith(url) || navItemOpen === idx
+                        ? "border-b-site-primary border-b-4 bg-interaction-sub"
+                        : ""
+                    }`}
+                    onClick={() => {
+                      if (navItemOpen === idx) {
+                        setNavItemOpen(-1)
+                      } else {
+                        setNavItemOpen(idx)
                       }
-                      return (
-                        <LinkComponent
-                          href={link.url}
-                          className="inline-flex items-center border-b-2 border-transparent px-1 pt-1 text-sm font-medium text-gray-500 hover:border-gray-300 hover:text-gray-700"
+                    }}
+                  >
+                    {name}
+                    {navItemOpen !== idx && (
+                      <BiChevronDown className="inline ml-1 -mt-1 text-2xl" />
+                    )}
+                    {navItemOpen === idx && (
+                      <BiChevronUp className="inline ml-1 -mt-1 text-2xl" />
+                    )}
+                  </button>
+                  <div
+                    className={`${
+                      navItemOpen === idx ? "absolute" : "hidden"
+                    } bg-white left-0 w-full px-4`}
+                  >
+                    <div className="flex flex-col mx-auto w-full max-w-screen-xl py-12 max-h-[32rem] overflow-auto">
+                      <div className="flex flex-row justify-between items-start pb-4 border-b-divider-medium border-b">
+                        <div className="flex flex-col gap-1">
+                          <h6 className="text-lg font-medium uppercase">
+                            {name}
+                          </h6>
+                          <p>{description}</p>
+                        </div>
+
+                        <button
+                          onClick={() => setNavItemOpen(-1)}
+                          aria-label="Close navigation item"
                         >
-                          {link.name}
-                        </LinkComponent>
-                      )
-                    })}
+                          Close
+                          <BiX className="inline ml-1 -mt-0.5 text-2xl" />
+                        </button>
+                      </div>
+
+                      <ul className="flex flex-row flex-wrap pt-12 gap-x-36 gap-y-8">
+                        {items.map((subItem) => (
+                          <li key={subItem.name} className="w-2/5">
+                            <div className="flex flex-col gap-1">
+                              <LinkComponent
```

Comment:
can we put the props given to `linkComponent` here as the type

Reply:
Not sure what you mean here, the `LinkComponent` is effectively the `<a>` tag. The actual typings of `LinkComponent` will be fixed later.

Reply:
Ok sgtm